app/models/merged_search_service.rb

```diff
@@ -0,0 +1,241 @@
+require 'digest'
+
+# Orchestrates merged "all" tab searches across Primo and TIMDEX.
+#
+# Handles parallel fetches, per-query totals caching, pagination calculation via
+# `MergedSearchPaginator`, and assembly of a controller-friendly response hash.
+class MergedSearchService
+  # Time to live value for cache expiration.
+  TTL = 10.minutes
+
+  # Initialize a new MergedSearchService.
+  #
+  # @param enhanced_query [Hash] query hash produced by `Enhancer`
+  # @param active_tab [String] the currently active tab (e.g. 'all')
+  # @param cache [Object] optional cache store responding to `read`/`write` (defaults to `Rails.cache`)
+  # @param primo_fetcher [#call] optional callable used to fetch Primo results; should accept `offset:, per_page:, query:`
+  # @param timdex_fetcher [#call] optional callable used to fetch TIMDEX results; should accept `offset:, per_page:, query:`
+  def initialize(enhanced_query:, active_tab:, cache: Rails.cache, primo_fetcher: nil, timdex_fetcher: nil)
+    @enhanced_query = enhanced_query
+    @active_tab = active_tab
+    @cache = cache
+    @primo_fetcher = primo_fetcher || method(:default_primo_fetch)
```

Comment:
It's unclear when/why we are injecting versus using the default methods.

The controller seems to be injecting so I'm not sure when the defaults are used.

Reply:
It's still unclear to me if we ever use the default path or if we are always injecting... it seems like the Controller and Tests both inject so is default never used?